avocado/utils/process.py

```diff
@@ -268,29 +268,30 @@ class CmdResult(object):
     """
 
     def __init__(self, command="", stdout_bytes=b"", stderr_bytes=b"",
-                 exit_status=None, duration=0, pid=None):
+                 exit_status=None, duration=0, encoding='utf-8', pid=None):
         self.command = command
         self.exit_status = exit_status
         self.stdout_bytes = stdout_bytes
         self.stderr_bytes = stderr_bytes
         self._stdout = None
         self._stderr = None
         self.duration = duration
+        self.encoding = encoding
```

Comment:
Apart from the fact we'll need to update the docstrings in the final version, this looks good.